pom.xml

```diff
@@ -33,16 +33,31 @@
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
       <version>${mavenVersion}</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-model</artifactId>
       <version>${mavenVersion}</version>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
       <version>${mavenVersion}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-artifact</artifactId>
+      <version>${mavenVersion}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-compat</artifactId>
+      <version>${mavenVersion}</version>
+      <scope>provided</scope>
     </dependency>
```

Comment:
Can you give some more context of why these were added?

Reply:
Oh, I see. Because it was in the list of dependencies you got a warning about.

Reply:
These are transitive dependencies that shouldn't be necessary to explicitly declare. After commenting them out, `mvnw dependency:tree` shows
```
[INFO] +- org.apache.maven:maven-plugin-api:jar:3.0:provided
[INFO] |  +- org.apache.maven:maven-artifact:jar:3.0:compile
```
and
```
[INFO] +- org.apache.maven.shared:file-management:jar:3.0.0:compile
[INFO] |  \- org.apache.maven.shared:maven-shared-io:jar:3.0.0:compile
[INFO] |     +- org.apache.maven:maven-compat:jar:3.0:compile
```

Reply:
if the file-management is updated to 3.1.0 that would not have been needed but the gmavenplus code currently is passing a log that doesn't resolve as needed.  So it seemed easier for a quicker pass on it to just override those to make maven happy for time being.  That next file management doesn't contain those btw.  And for whatever reason a build locally for me fails at the current master before this and I didnt' have time to look into that when GHA were completely fine.